tests/test_io/test_mongodb.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+
+"""Tests for the jsonschema node validation."""
+
+from copy import deepcopy
+from collections import namedtuple
+from typing import Any, List, Mapping, Optional, Tuple
+import unittest
+
+from pymongo import MongoClient
+from pymongo.collection import Collection
+from pymongo.errors import ConnectionFailure
+
+from pybel import BELGraph, to_sbel
+from pybel.dsl import Entity, ComplexAbundance, Gene, Protein, Rna
+from pybel.io.mongodb import (
+    _entity_to_dict,
+    _rm_mongo_keys,
+    to_mongodb,
+    find_nodes,
+    get_edges_from_node,
+    get_edges_from_criteria
+)
+from pybel.testing.utils import n as n_
+
+g1 = Gene('hgnc', '1')
+r1 = Rna('hgnc', '1')
+p1 = Protein('hgnc', '1')
+g2 = Gene('hgnc', '2')
+r2 = Rna('hgnc', '2')
+p2 = Protein('hgnc', '2')
+g3 = Gene('hgnc', '3')
+p3 = Protein('hgnc', '3')
+
+ca = ComplexAbundance([p1, g2], name='ca', namespace='hgnc')
+
+# Define a namedtuple to make it easier to pass around important node information
+NodeInfo = namedtuple('NodeInfo', 'name identifier function variants')
+
+
+def id_info(node: Mapping[str, Any]) -> NodeInfo:
```

Comment:
I don't see why you need the NodeInfo object, can't you access all of the name, identifier, function, and variants directly from the node object anyway?

Reply:
Yep I can, but having the NodeInfo object reduces the repetition of .get() calls on node objects and makes it easier to add criteria (like the function param I added yesterday). Does that make it worth keeping, or does it just add clutter/complexity?